### PR TITLE
SQL Scripts Re-based

### PR DESCRIPTION
**Pull request by LtRipley36706**
Hopefully this should bring the base scripts into alignment and trim out unnecessary tables from each.

### CODE REVIEW

File: Database/Base/ShardBase.sql [hunk 1]
```diff
@@ -251,122 +438,138 @@ CREATE TABLE `ace_position` (
...
- `physicsDescriptionFlag` int(10) unsigned ,
- `weenieHeaderFlags` int(10) unsigned ,
- `itemType` int(10) unsigned 
-)*/;
-
-/*Table structure for table `vw_ace_object` */
+/*!50001 DROP VIEW IF EXISTS `vw_ace_character`*/;
+SET @saved_cs_client     = @@character_set_client;
+SET character_set_client = utf8;
+/*!50001 CREATE VIEW `vw_ace_character` AS SELECT
```
Comment (Mogwai-TheFurry) [inline Database/Base/ShardBase.sql]: if we're making the view useless, can we just remove it?

File: Database/Base/ShardBase.sql [hunk 2]
```diff
@@ -251,122 +438,138 @@ CREATE TABLE `ace_position` (
...
-
-/*!50001 DROP TABLE IF EXISTS `vw_teleport_location` */;
-/*!50001 DROP VIEW IF EXISTS `vw_teleport_location` */;
-
-/*!50001 CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `vw_teleport_location` AS (select `apoi`.`name` AS `name`,`ap`.`landblockRaw` AS `landblock`,`ap`.`posX` AS `posX`,`ap`.`posY` AS `posY`,`ap`.`posZ` AS `posZ`,`ap`.`qW` AS `qW`,`ap`.`qX` AS `qX`,`ap`.`qY` AS `qY`,`ap`.`qZ` AS `qZ` from (`ace_poi` `apoi` join `ace_position` `ap` on(`apoi`.`positionId` = `ap`.`positionId`)) where `ap`.`positionType` = 28) */;
+/*!50001 DROP VIEW IF EXISTS `vw_ace_object`*/;
+SET @saved_cs_client     = @@character_set_client;
+SET character_set_client = utf8;
+/*!50001 CREATE VIEW `vw_ace_object` AS SELECT 
+ 1 AS `aceObjectId`,
```
Comment (Mogwai-TheFurry) [inline Database/Base/ShardBase.sql]: if we're making the view useless, can we just remove it?

File: Database/Base/ShardBase.sql [hunk 3]
```diff
@@ -251,122 +438,138 @@ CREATE TABLE `ace_position` (
...
... `accountId`,`aops`.`propertyValue` AS `NAME`,`aopb`.`propertyValue` AS `deleted`,`aopbi`.`propertyValue` AS `deleteTime`,`ao`.`weenieClassId` AS `weenieClassId`,`awc`.`weenieClassDescription` AS `weenieClassDescription`,`ao`.`aceObjectDescriptionFlags` AS `aceObjectDescriptionFlags`,`ao`.`physicsDescriptionFlag` AS `physicsDescriptionFlag`,`ao`.`weenieHeaderFlags` AS `weenieHeaderFlags`,`aopi`.`propertyValue` AS `itemType` from ((((((`ace_object` `ao` join `ace_weenie_class` `awc` on((`ao`.`weenieClassId` = `awc`.`weenieClassId`))) join `ace_object_properties_string` `aops` on(((`ao`.`aceObjectId` = `aops`.`aceObjectId`) and (`aops`.`strPropertyId` = 1)))) join `ace_object_properties_bool` `aopb` on(((`ao`.`aceObjectId` = `aopb`.`aceObjectId`) and (`aopb`.`boolPropertyId` = 9001)))) join `ace_object_properties_int` `aopi` on(((`ao`.`aceObjectId` = `aopi`.`aceObjectId`) and (`aopi`.`intPropertyId` = 1)))) join `ace_object_properties_bigint` `aopbi` on(((`ao`.`aceObjectId` = `aopbi`.`aceObjectId`) and (`aopbi`.`bigIntPropertyId` = 9001)))) join `ace_object_properties_iid` `aopiidacc` on(((`ao`.`aceObjectId` = `aopiidacc`.`aceObjectId`) and (`aopiidacc`.`iidPropertyId` = 9001)))) */;
```
Comment (Mogwai-TheFurry) [inline Database/Base/ShardBase.sql]: remove comment block garbage.  script should fail for substandard versions.

File: Database/Base/ShardBase.sql [hunk 4]
```diff
@@ -251,122 +438,138 @@ CREATE TABLE `ace_position` (
...
+/*!50013 DEFINER=`root`@`localhost` SQL SECURITY DEFINER */
+/*!50001 VIEW `vw_ace_object` AS select `ao`.`aceObjectId` AS `aceObjectId`,`aops`.`propertyValue` AS `name`,`ao`.`weenieClassId` AS `weenieClassId`,`awc`.`weenieClassDescription` AS `weenieClassDescription`,`ao`.`aceObjectDescriptionFlags` AS `aceObjectDescriptionFlags`,`ao`.`physicsDescriptionFlag` AS `physicsDescriptionFlag`,`ao`.`weenieHeaderFlags` AS `weenieHeaderFlags`,`aopi`.`propertyValue` AS `itemType`,`ap`.`positionId` AS `positionId`,`ap`.`positionType` AS `positionType`,`ap`.`landblockRaw` AS `LandblockRaw`,`ap`.`posX` AS `posX`,`ap`.`posY` AS `posY`,`ap`.`posZ` AS `posZ`,`ap`.`qW` AS `qW`,`ap`.`qX` AS `qX`,`ap`.`qY` AS `qY`,`ap`.`qZ` AS `qZ` from ((((`ace_object` `ao` join `ace_weenie_class` `awc` on((`ao`.`weenieClassId` = `awc`.`weenieClassId`))) join `ace_object_properties_string` `aops` on(((`ao`.`aceObjectId` = `aops`.`aceObjectId`) and (`aops`.`strPropertyId` = 1)))) join `ace_object_properties_int` `aopi` on(((`ao`.`aceObjectId` = `aopi`.`aceObjectId`) and (`aopi`.`intPropertyId` = 1)))) join `ace_position` `ap` on(((`ao`.`aceObjectId` = `ap`.`aceObjectId`) and (`ap`.`positionType` = 1)))) */;
```
Comment (Mogwai-TheFurry) [inline Database/Base/ShardBase.sql]: booby-trap comment blocks again

File: Database/Base/WorldBase.sql [hunk 5]
```diff
@@ -1,397 +1,420 @@
...
-  `currentMotionState` TEXT DEFAULT NULL,
+  `aceObjectId` int(10) unsigned NOT NULL,
+  `aceObjectDescriptionFlags` int(10) unsigned NOT NULL,
+  `weenieClassId` int(10) unsigned NOT NULL,
+  `weenieHeaderFlags` int(10) unsigned DEFAULT NULL,
+  `physicsDescriptionFlag` int(10) unsigned DEFAULT NULL,
+  `currentMotionState` text DEFAULT NULL,
   PRIMARY KEY (`aceObjectId`),
   KEY `idx_weenie` (`weenieClassId`),
   CONSTRAINT `fk_weenie_ao` FOREIGN KEY (`weenieClassId`) REFERENCES `ace_weenie_class` (`weenieClassId`)
```
Comment (Mogwai-TheFurry) [inline Database/Base/WorldBase.sql]: can't create a foreign key to ace_weenie_class before you create the table.  please test script in a blank DB and ensure it runs fully on the first attempt.

File: Database/Base/WorldBase.sql [hunk 6]
```diff
@@ -1,397 +1,420 @@
...
+--
 
-/*!50001 DROP TABLE IF EXISTS `vw_teleport_location` */;
-/*!50001 DROP VIEW IF EXISTS `vw_teleport_location` */;
+DROP TABLE IF EXISTS `vw_ace_object`;
+/*!50001 DROP VIEW IF EXISTS `vw_ace_object`*/;
+SET @saved_cs_client     = @@character_set_client;
+SET character_set_client = utf8;
+/*!50001 CREATE VIEW `vw_ace_object` AS SELECT 
+ 1 AS `aceObjectId`,
```
Comment (Mogwai-TheFurry) [inline Database/Base/WorldBase.sql]: if we're making the view useless, can we just remove it?

File: Database/Base/WorldBase.sql [hunk 7]
```diff
@@ -1,397 +1,420 @@
...
+-- Temporary view structure for view `vw_teleport_location`
+--
 
-/*!50001 CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `vw_teleport_location` AS (select `apoi`.`name` AS `name`,`ap`.`landblockRaw` AS `landblock`,`ap`.`posX` AS `posX`,`ap`.`posY` AS `posY`,`ap`.`posZ` AS `posZ`,`ap`.`qW` AS `qW`,`ap`.`qX` AS `qX`,`ap`.`qY` AS `qY`,`ap`.`qZ` AS `qZ` from (`ace_poi` `apoi` join `ace_position` `ap` on(`apoi`.`positionId` = `ap`.`positionId`)) where `ap`.`positionType` = 28) */;
+DROP TABLE IF EXISTS `vw_teleport_location`;
+/*!50001 DROP VIEW IF EXISTS `vw_teleport_location`*/;
+SET @saved_cs_client     = @@character_set_client;
+SET character_set_client = utf8;
+/*!50001 CREATE VIEW `vw_teleport_location` AS SELECT 
+ 1 AS `name`,
```
Comment (Mogwai-TheFurry) [inline Database/Base/WorldBase.sql]: remove

File: Database/Base/WorldBase.sql [hunk 8]
```diff
@@ -1,397 +1,420 @@
...
...elect `ao`.`aceObjectId` AS `aceObjectId`,`aops`.`propertyValue` AS `name`,`ao`.`weenieClassId` AS `weenieClassId`,`ao`.`currentMotionState` AS `currentMotionState`,`awc`.`weenieClassDescription` AS `weenieClassDescription`,`ao`.`aceObjectDescriptionFlags` AS `aceObjectDescriptionFlags`,`ao`.`physicsDescriptionFlag` AS `physicsDescriptionFlag`,`ao`.`weenieHeaderFlags` AS `weenieHeaderFlags`,`aopi`.`propertyValue` AS `itemType`,`ap`.`positionId` AS `positionId`,`ap`.`positionType` AS `positionType`,`ap`.`landblockRaw` AS `LandblockRaw`,`ap`.`landblock` AS `landblock`,`ap`.`cell` AS `cell`,`ap`.`posX` AS `posX`,`ap`.`posY` AS `posY`,`ap`.`posZ` AS `posZ`,`ap`.`qW` AS `qW`,`ap`.`qX` AS `qX`,`ap`.`qY` AS `qY`,`ap`.`qZ` AS `qZ` from ((((`ace_object` `ao` join `ace_weenie_class` `awc` on((`ao`.`weenieClassId` = `awc`.`weenieClassId`))) join `ace_object_properties_string` `aops` on(((`ao`.`aceObjectId` = `aops`.`aceObjectId`) and (`aops`.`strPropertyId` = 1)))) join `ace_object_properties_int` `aopi` on(((`ao`.`aceObjectId` = `aopi`.`aceObjectId`) and (`aopi`.`intPropertyId` = 1)))) join `ace_position` `ap` on(((`ao`.`aceObjectId` = `ap`.`aceObjectId`) and (`ap`.`positionType` = 1)))) */;
```
Comment (Mogwai-TheFurry) [inline Database/Base/WorldBase.sql]: comment block garbage again

File: Database/Base/WorldBase.sql [hunk 9]
```diff
@@ -1,397 +1,420 @@
...
+/*!50001 DROP VIEW IF EXISTS `vw_teleport_location`*/;
+/*!50001 SET @saved_cs_client          = @@character_set_client */;
+/*!50001 SET @saved_cs_results         = @@character_set_results */;
+/*!50001 SET @saved_col_connection     = @@collation_connection */;
+/*!50001 SET character_set_client      = utf8 */;
+/*!50001 SET character_set_results     = utf8 */;
+/*!50001 SET collation_connection      = utf8_general_ci */;
+/*!50001 CREATE ALGORITHM=UNDEFINED */
+/*!50013 DEFINER=`root`@`localhost` SQL SECURITY DEFINER */
+/*!50001 VIEW `vw_teleport_location` AS (select `apoi`.`name` AS `name`,`ap`.`landblockRaw` AS `landblock`,`ap`.`posX` AS `posX`,`ap`.`posY` AS `posY`,`ap`.`posZ` AS `posZ`,`ap`.`qW` AS `qW`,`ap`.`qX` AS `qX`,`ap`.`qY` AS `qY`,`ap`.`qZ` AS `qZ` from (`ace_poi` `apoi` join `ace_position` `ap` on((`apoi`.`positionId` = `ap`.`positionId`))) where (`ap`.`positionType` = 28)) */;
```
Comment (Mogwai-TheFurry) [inline Database/Base/WorldBase.sql]: comment block garbage again

File: Database/Base/ShardBase.sql (hunk 1 above)
Comment (LtRipley36706) [inline Database/Base/ShardBase.sql]: Did not test removing it, will do so.

File: Database/Base/ShardBase.sql (hunk 2 above)
Comment (LtRipley36706) [inline Database/Base/ShardBase.sql]: Did not test removing view, will do so.

File: Database/Base/ShardBase.sql (hunk 3 above)
Comment (LtRipley36706) [inline Database/Base/ShardBase.sql]: This works for mysql and is created by mysqldump. Tooling does not allow for disables.

File: Database/Base/ShardBase.sql (hunk 4 above)
Comment (LtRipley36706) [inline Database/Base/ShardBase.sql]: This works for mysql and is created by mysqldump. Tooling does not allow for disables.

File: Database/Base/WorldBase.sql (hunk 5 above)
Comment (LtRipley36706) [inline Database/Base/WorldBase.sql]: This works for mysql and is created by mysqldump. I do test on blank DBs and it executes without fail. Tooling does not allow for order assignment, FK checks are turned off and it works.

File: Database/Base/WorldBase.sql (hunk 6 above)
Comment (LtRipley36706) [inline Database/Base/WorldBase.sql]: Did not test removing view, will do so.

File: Database/Base/WorldBase.sql (hunk 7 above)
Comment (LtRipley36706) [inline Database/Base/WorldBase.sql]: Did not test removing view, will do so.

File: Database/Base/WorldBase.sql (hunk 8 above)
Comment (LtRipley36706) [inline Database/Base/WorldBase.sql]: This works for mysql and is created by mysqldump. Tooling does not allow for disables.

File: Database/Base/WorldBase.sql (hunk 9 above)
Comment (LtRipley36706) [inline Database/Base/WorldBase.sql]: This works for mysql and is created by mysqldump. Tooling does not allow for disables.

**Pull request merged**